src/linear_algebra/bilinear_form.lean

```diff
@@ -400,3 +410,175 @@ begin
 end
 
 end alt_bilin_form
+
+namespace bilin_form
+
+section linear_adjoints
+
+variables {R : Type u} [comm_ring R]
+variables {M : Type v} [add_comm_group M] [module R M]
+variables {M₂ : Type v} [add_comm_group M₂] [module R M₂]
+variables (B B' : bilin_form R M) (B₂ : bilin_form R M₂)
+variables (f f' : M →ₗ[R] M₂) (g g' : M₂ →ₗ[R] M)
+
+/-- Given a pair of modules equipped with bilinear forms, this is the condition for a pair of
+maps between them to be mutually adjoint. -/
+def is_adjoint_pair := ∀ x y, B₂ (f x) y = B x (g y)
+
```

Comment:
I would add a lemma here
```suggestion

lemma is_adjoint_pair.eq (h : is_adjoint_pair B B2 f g) :
  ∀ {x y}, B₂ (f x) y = B x (g y) := h

```
This way, if you have an assumption `h : is_adjoint_pair`, then you can very easily rewrite from left to right and vice versa, using projection notation `h.eq`.
I think you will need this if you switch to the `{{ ... }}` form above.